File: server/src/main/java/ftn/booking/model/Complaint.java
package ftn.booking.model;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;


@Entity
@Setter
@Getter
public class Complaint {

    @Id
    @GeneratedValue( strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    private String content;

    @ManyToOne
    private Client client;

    @ManyToOne
    private User user;

    @ManyToOne
    private Boat boat;

    @ManyToOne
    private Cottage cottage;

    @ManyToOne
    private Adventure adventure;


}

File: client/src/app/dto/complaint.ts
import { Adventure } from "./adventure";
import { Boat } from "./boat";
import { Client } from "./client";
import { Cottage } from "./cottage";


export class Complaint 
{
    id !: Number
    name !: String
    content !: String
    client !: Client
    boat !: Boat
    cottage !: Cottage
    adventure !: Adventure
}
